core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -387,13 +392,68 @@ long getValueReference(int entryIndex) {
         return getEntryFieldLong(entryIndex, OFFSET.VALUE_REFERENCE);
     }
 
+    long getValueReferenceAndVersion(int entryIndex, int[] version) {
+        long valueReference;
+        int v;
+        do {
+            v = getValueVersion(entryIndex);
+            valueReference = getValueReference(entryIndex);
+        } while (v != getValueVersion(entryIndex));
+        version[0] = v;
+        return valueReference;
+    }
+
+    int completeLinking(LookUp lookUp) {
+        int entryVersion = lookUp.version;
+        // no need to complete a thing
+        if (entryVersion > INVALID_VERSION) {
+            return entryVersion;
+        }
+        if (!publish()) {
```

Comment:
Publish without parameters... It is already so far from what we had... Why to publish if nobody can help? It is not the problem of this code I am just frustrated... I hope I will be able to make the rebalance non-blocking again...